crates/orderbook/src/database/orders.rs

```diff
@@ -221,23 +225,45 @@ async fn insert_order(order: &Order, ex: &mut PgConnection) -> Result<(), Insert
 }
 
 async fn insert_quote(
-    uid: &OrderUid,
+    order: &Order,
     quote: &Quote,
     ex: &mut PgConnection,
 ) -> Result<(), InsertionError> {
-    let quote = database::orders::Quote {
-        order_uid: ByteArray(uid.0),
+    let dbquote = database::orders::Quote {
+        order_uid: ByteArray(order.metadata.uid.0),
         gas_amount: quote.data.fee_parameters.gas_amount,
         gas_price: quote.data.fee_parameters.gas_price,
         sell_token_price: quote.data.fee_parameters.sell_token_price,
         sell_amount: u256_to_big_decimal(&quote.sell_amount),
         buy_amount: u256_to_big_decimal(&quote.buy_amount),
         solver: ByteArray(quote.data.solver.0),
+        verified: quote.data.verified,
     };
-    database::orders::insert_quote(ex, &quote)
+    let dbinteractions = quote
+        .data
+        .interactions
+        .iter()
+        .enumerate()
+        .map(|(idx, interaction)| {
+            Ok(OrderQuoteInteraction {
+                order_uid: dbquote.order_uid,
+                index: idx.try_into()?,
```

Comment:
I think it's unfortunate to add a new very specific error case for something that will never happen in practice.
If we need to introduce a new error type I would make it less specific so it can at least be used by all the error cases that will never happen in practice. Alternatively we could also unwrap here.

Reply:
I was thinking which approach would be better and decided to add new error variant, but I think the better option here would be to use unwrap here.